packages/node/src/nodeClient.ts

```diff
@@ -43,39 +66,63 @@ export class NodeClient implements Client<Options> {
     }
 
     this._annotateEvent(event);
+    // Add event to unsent events queue.
+    this._events.push(event);
 
-    const payload = JSON.stringify({
-      api_key: this._apiKey,
-      events: [event],
-    });
-
-    const requestOptions = {
-      hostname: AMPLITUDE_API_HOST,
-      path: AMPLITUDE_API_PATH,
-      method: 'POST',
-      headers: {
-        'Content-Type': 'application/json',
-      },
-    };
-
-    const req = https.request(requestOptions, res => {
-      res.on('data', _ => {
-        // Request finishes.
-        // We currently don't have error handling or retry, but we will add it soon.
-      });
-    });
+    const bufferLimit = this._options.maxCachedEvents ?? 100;
```

Comment:
if we use null coalescence, we should add a node engine property to the package.json bc it's only supported on node 14

Otherwise, just stick to the or operand (`||`)

Reply:
node engine properties? What is it? Sorry I am too new to JS ecosystem.

Reply:
no worries! https://docs.npmjs.com/files/package.json#engines

Reply:
We shouldn't have to worry about engine at this level because this is TypeScript and will be transpiled to JavaScript by the TS compiler.

So we instead need to be conscious of what the configured compiler target is and set an engine requirement appropriate for the build library.

We should probably be aiming for a slightly older Node engine for minimum support. I think 10 would be the absolutely minimum to be useful to a wide audience. Stretching for 8 might be good but I'm not sure what the trade-offs would be for the built code.